### PR TITLE
Added parsing of Turbulence and Icing

### LINKED ISSUE #413: Wind speed reported incorrectly
**Describe the bug**

Wind speed is reported incorrectly.

**To Reproduce**

Using the following TAF:

```
TAF KLSV 222300Z 2223/2405 21020G35KT 8000 BLDU BKN160 530009 QNH2941INS
  TEMPO 2223/2302 23035G52KT BKN150 560009
  BECMG 2305/2306 35015G25KT 9999 VCSH BKN140 520009 QNH2948INS
  BECMG 2316/2317 34010G18KT 9999 NSW SCT170 QNH2969INS TX28/2323Z TN12/2314Z
```

See wind speed = 0 for the initial TAF conditions, and the first BECMG.

image

**Expected behavior**

A valid wind speed, see this screenshot for aviationweather.gov's interpretation:

image

**Additional context**

Appears to be incorrectly matching the turbulence part of the TAF - `530009`.

More info page 29 on turbulence part of TAF http://prnfc.org/wp-content/uploads/2016/12/AF-METAR-TAF-Codes.pdf:

image

Thanks you! :)

### PR DESCRIPTION
**Pull request by mivek**
This PR fixes #413.

Updated regex of Wind
Added parsing of Turbulence and Icing for TAF

### CODE REVIEW

**Comment by aeharding** [comment]
Hi @mivek, thanks so much for doing this! However, I believe icing/turbulence can also be a trend property, and also there can be multiple icing/turbulence commands per group. For example the following (crazy) TAF has multiple icing/turbulence groups, sometimes for the same trend:

```
TAF AMD KNID 0115/0215 21006KT 9999 SCT250 QNH2981INS
  BECMG 0116/0118 19014G22KT 9999 FEW120 SCT250 520009 520909 QNH2978INS WND 160V230
  BECMG 0118/0120 19018G26KT 9000 BLDU FEW003 FEW120 SCT250 530009 530909 QNH2972INS
  BECMG 0120/0122 19022G35KT 9000 BLDU FEW003 SCT120 BKN250 560009 560906 531509 QNH2967INS
  TEMPO 0122/0202 19030G40KT 1600 BLDUSA BKN002 SCT120 BKN250
  BECMG 0202/0204 20018G30KT 9999 NSW SCT120 BKN220 530008 530909 QNH2972INS
  BECMG 0208/0210 21015G25KT 9999 BKN120 BKN220 611208 530009 540909 QNH2974INS
  BECMG 0214/0215 22012G25KT 9999 BKN120 BKN220 611208 520009 540909 QNH2980INS T23/0122Z T13/0213Z AMD 1540
```

Entire PDF of below image (aviationweather.gov parsing logic):

[KNID-multiple-turbulence.pdf](https://github.com/mivek/MetarParser/files/9913692/KNID-multiple-turbulence.pdf)

image

File: metarParser-parsers/src/main/java/io/github/mivek/command/taf/TurbulenceCommand.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import java.util.regex.Pattern;
+
+/**
+ * Handle the parsing of the Turbulence part of a TAF.
+ * @author Jean-Kevin KPADEY
+ */
+public final class TurbulenceCommand implements Command {
+
+  /** Regex pattern for the turbulence. */
+  private static final Pattern TURBULENCE_PATTERN = Pattern.compile("^5(\\d)(\\d{3})(\\d)$");
```
Comment (aeharding) [inline metarParser-parsers/src/main/java/io/github/mivek/command/taf/TurbulenceCommand.java:16]: The intensity could also be "X"

**Comment by mivek** [comment]
Thank you for the feedback, 
So here is the solution I propose to also have the icing and the turbulence in the Trends.
- I will move the attributes icing and turbulence to the class [AbstractWeatherContainer](https://github.com/mivek/MetarParser/blob/master/metarParser-entities/src/main/java/io/github/mivek/model/AbstractWeatherContainer.java), this class is the parent class of trends, METAR and TAF object. 
- Transform the attribute to list
- Move the commands IcingCommand and TurbulenceCommand in the package `io.github.mivek.common` and add the commands before the WindShearCommand inside the [CommonCommandSupplier](https://github.com/mivek/MetarParser/blob/master/metarParser-parsers/src/main/java/io/github/mivek/command/common/CommonCommandSupplier.java).
- Revert the changes made in TAFParser.

This also makes turbulences and icing available in the METAR object. I could throw an error if a call to getTurbulence or getIcing is made from a metar object but I do not think the parser should be this strict.

I will try to push something over the weekend

**Comment by aeharding** [comment]
That makes sense @mivek.

Regarding preventing it from showing up on metars, what I did in the web port is created two commands (icing and turbulence) and a TAF command parser for those two commands (similar to how METAR have their own unique commands for runway conditions, altimeter, etc)

https://github.com/aeharding/metar-taf-parser/pull/39/files

https://github.com/aeharding/metar-taf-parser/blob/82aedb327f13fe891cbbb5073935169ff7445cea/src/command/taf.ts

I integrated the TAF command supplier into the parser here: https://github.com/aeharding/metar-taf-parser/pull/39/files#diff-148bacb6839f9d7e10901515876440d2e966036b4b9ff4ef8562349353df35d1

**Comment by aeharding** [comment]
I should also note that instead of putting it on the AbstractWeatherContainer (which metars inherit from), I created a new intermediate for TAFs and TAF trends called `ITAFGroups` (it's not the best name...) which has icing and turbulence. Again, just another option :)

https://github.com/aeharding/metar-taf-parser/pull/39/files#diff-58a5f4a5424e479db505f909fa639aea8fbb996324bc73430934932455134dff

**Comment by aeharding** [comment]
Sorry, one last thought, a nice part about making a separate command supplier for TAF specific commands (such as icing/turbulence) is that the TAF-specific altimeter group could be added in the future:

Page 1-24: https://www.t6bdriver.com/uploads/6/4/7/7/64775059/p-770_aviation_weather_student_study_guide.pdf

Page 30: http://prnfc.org/wp-content/uploads/2016/12/AF-METAR-TAF-Codes.pdf#page=30

File: metarParser-entities/src/main/java/io/github/mivek/model/ITafGroups.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+  /**
+   * @param turbulence The turbulence to add.
+   */
+  default void addTurbulence(Turbulence turbulence) {
+    getTurbulences().add(turbulence);
+  }
+
+  /**
+   * @return The list of icing elements.
```
[the remark this answers is not in the document]
Comment (mivek) [reply metarParser-entities/src/main/java/io/github/mivek/model/ITafGroups.java:23]: @sonatype-lift ignoreall

**Pull request merged**